datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -251,7 +251,9 @@ pub fn out_ordering(
     func: &FuncMonotonicity,
     arg_orderings: &[SortProperties],
 ) -> SortProperties {
-    func.iter().zip(arg_orderings).fold(
+    let monotonicity_vec: Vec<Option<bool>> = func.into();
+
+    monotonicity_vec.iter().zip(arg_orderings).fold(
```

Comment:
This logic will unfortunately get a little complicated given the issue with `From`.